typescript/cli/src/commands/warp.ts

```diff
@@ -427,6 +448,236 @@ export const check: CommandModuleWithContext<{
+        // This will probably never fail, but allows the type to be inferred as not undefined.
+        assert(origin, '--origin is required');
+        assert(destination, '--destination is required');
+        assert(amount, '--amount is required');
+
+        warnYellow(
+          `Manual rebalance strategy selected. Origin: ${origin}, Destination: ${destination}, Amount: ${amount}`,
+        );
+
+        const warpCore = contextFactory.getWarpCore();
+        const rebalancer = contextFactory.createRebalancer();
+        const originToken = warpCore.tokens.find((t) => t.chainName === origin);
+
+        try {
+          await rebalancer.rebalance([
+            {
+              origin,
+              destination,
+              amount: BigInt(toWei(amount, originToken!.decimals)),
+            },
+          ]);
+          logGreen(
+            `✅ Manual rebalance from ${origin} to ${destination} for amount ${amount} submitted successfully.`,
+          );
+          process.exit(0);
+        } catch (e) {
+          errorRed(
+            `❌ Manual rebalance from ${origin} to ${destination} failed.`,
+          );
+          errorRed(format(e));
+          process.exit(1);
+        }
+      }
+
+      // Instantiates the monitor that will observe the warp route
+      monitor = contextFactory.createMonitor();
+
+      // Instantiates the metrics that will publish stats from the monitored data
+      metrics = withMetrics ? await contextFactory.createMetrics() : undefined;
+
+      // Instantiates the strategy that will compute how rebalance routes should be performed
+      strategy = await contextFactory.createStrategy(metrics);
+
+      // Instantiates the rebalancer in charge of executing the rebalancing transactions
+      rebalancer = !rebalancerConfig.monitorOnly
+        ? contextFactory.createRebalancer(metrics)
+        : undefined;
+
+      if (rebalancerConfig.monitorOnly) {
+        warnYellow(
+          'Running in monitorOnly mode: no transactions will be executed.',
+        );
+      }
+
+      if (withMetrics) {
+        warnYellow(
+          'Metrics collection has been enabled and will be gathered during execution',
+        );
+
+        // Initialize execution status metrics, if metrics are enabled
+        metrics?.initializeRebalancerMetrics();
+      }
+    } catch (e) {
+      errorRed('Rebalancer startup error:', format(e));
+      process.exit(1);
+    }
+
+    try {
+      // Setup monitor event listeners before starting it.
+      // These handlers deal with events and errors occurring *after* the monitor has successfully started.
+      monitor
+        // Observe balances events and process rebalancing routes
+        .on(MonitorEventType.TokenInfo, (event) => {
+          if (metrics) {
+            for (const tokenInfo of event.tokensInfo) {
+              metrics.processToken(tokenInfo).catch((e) => {
```

Comment:
is this `.catch` needed, given that processToken does not rethrow i.e. uses `tryFn()`?